soepy/test/random_init.py

```diff
@@ -94,9 +94,29 @@ def random_init(constr=None):
         "PARAMETERS"
     ]["delta_s3"] = np.random.uniform(0.1, 0.9, 3).tolist()
 
-    init_dict["PARAMETERS"]["theta_p"], init_dict["PARAMETERS"][
-        "theta_f"
-    ] = np.random.uniform(0.5, 5, 2).tolist()
+    # Random number of types: 1, 2, 3, 4, or 5
+    num_types = 1
+    # num_types = int(np.random.choice([1, 2, 3, 4, 5], 1))
+
+    # One type: ensure share equals 1
+    if num_types == 1:
+        init_dict["PARAMETERS"]["theta_p1"], init_dict["PARAMETERS"][
```

Comment:
lines 103-106 are rather repeated in line 112-119, please try to merge

Reply:
It seems like it, but in order to assign the integer indices 1 in case of one type and 1,2,3 etc. in case of many types, I do not see a good way of not keeping the if clause and the for loop separate.